src/nat/authentication/oauth2/oauth2_auth_code_flow_provider.py

```
@@ -74,26 +79,33 @@ async def _attempt_token_refresh(self, user_id: str, auth_result: AuthResult) ->
 
     def _set_custom_auth_callback(self,
                                   auth_callback: Callable[[OAuth2AuthCodeFlowProviderConfig, AuthFlowType],
-                                                          AuthenticatedContext]):
+                                                          Awaitable[AuthenticatedContext]]):
         self._auth_callback = auth_callback
 
     async def authenticate(self, user_id: str | None = None, **kwargs) -> AuthResult:
```

Ruff (0.13.2)

Unused method argument: `kwargs`

(ARG002)